cmd/mdatagen/internal/metadata/generated_metrics.go

```diff
@@ -281,6 +281,12 @@ func newMetricOptionalMetric(settings MetricSettings) metricOptionalMetric {
 	return m
 }
 
+// MetricsBuilderConfig is a structural subset of an otherwise 1-1 copy of metadata.yaml
+type MetricsBuilderConfig struct {
+	MetricsSettings            MetricsSettings            `mapstructure:"metrics,squash"`
+	ResourceAttributesSettings ResourceAttributesSettings `mapstructure:"resource_attributes,squash"`
```

Comment:
let's call the fields the same names as mapstructure. Also, please remove squash in them, that's wrong.
```suggestion
	Metrics            MetricsSettings            `mapstructure:"metrics"`
	ResourceAttributes ResourceAttributesSettings `mapstructure:"resource_attributes"`
```